ksmt-core/src/main/kotlin/org/ksmt/expr/rewrite/simplify/ArithSimplification.kt

```diff
@@ -54,6 +54,7 @@ fun <T : KArithSort> KContext.simplifyArithAdd(args: List<KExpr<T>>): KExpr<T> {
         }
     }
 
+    // If all args are constants we have no simplifiedArgs and a single constant value
```

Comment:
My question was more about `Why does it work in cases when the args list is empty?`, is it because if construction invariants?

Reply:
You mean that if the args are empty, the result will be zero, which is correct for addition?

Reply:
No 

```kotlin
    // assume that the list of arguments is empty

    val simplifiedArgs = ArrayList<KExpr<T>>(args.size) // this one is empty as well
    var constantTerm = RealValue.zero
    
    // we won't go inside of the cycle
    for (arg in args) {
        // flat one level
        if (arg is KAddArithExpr<T>) {
            for (flatArg in arg.args) {
                constantTerm = addArithTerm(constantTerm, flatArg, simplifiedArgs)
            }
        } else {
            constantTerm = addArithTerm(constantTerm, arg, simplifiedArgs)
        }
    }

    // this is satisfied
    if (simplifiedArgs.isEmpty()) {
         // args.first() will throw an exception
         return numericValue(constantTerm, args.first().sort)
```

Reply:
I added an explicit arguments checks